insights/specs/datasources/pcp.py

```diff
@@ -74,3 +91,38 @@ def pmlog_summary_args(broker):
         raise SkipComponent("Failed to get pmlogger metrics: {0}".format(str(e)))
 
     return "{0} {1}".format(pm_file, ' '.join(sorted(metrics)))
+
+
+@datasource(ros_collect, Hostname, HostContext)
+def pcp_raw_files(broker):
+    """
+    Return all existing paths of PCP RAW data files of yesterday.
+
+    .. note::
+        The "hostname" in the file paths will be removed by the "save_as"
+        in the `pcp_raw_data` spec.  All the collected PCP RAW data files
+        will be stored in "var/log/pcp/pmlogger" directory in the archive.
+
+    Returns:
+        list(str): list of file path
+
+    Raises:
+        ContentException: Raises when no such directory or number of PCP RAW
+                          files less than 3.
+    """
+    hn = broker[Hostname]
+    pm_root = os.path.join("/var/log/pcp/pmlogger/", hn.hostname)
+    if not os.path.exists(pm_root):
+        pm_root = os.path.join("/var/log/pcp/pmlogger/", hn.fqdn)
```

Comment:
I can see hn.hostname and hn.fqdn returns same output for me,
~~~
(Pdb) hn.hostname
'localhost'
(Pdb) hn.fqdn
'localhost'
~~~

However if I use hostnameDefault then I can see raw is returning the value which resembles to pcp directory naming convention,
~~~
(Pdb) hn_default.raw
'localhost.localdomain'
(Pdb) hn_default.hostname
'localhost'
~~~

Yesterday I missed to check the man page of hostname command. Below is the snippet from it, so yeah without any param hostname cmd uses the same call as pcp,
~~~
       When called without any arguments, the program displays the current names:
       hostname will print the name of the system as returned by the gethostname(2) function.
~~~

Considering this I have two thoughts,
1. We can blindly depend on hostnameDefault.raw as it resembles to what pcp is doing.
2. Maybe we should still try the both ways, I mean checking for short and long names to be safe.

I wish to go with 2nd option and use hostnameDefault, let me know your thoughts.

Thank you!

Reply:
Sure things